plugins/slipping-away-users/frontend/public/javascripts/countly.models.js

```diff
@@ -1,30 +1,106 @@
-/*global countlyCommon,jQuery*/
-(function(countlySlippingPlugin, $) {
-    var _data = {};
-    /**
-     * This is for initializing model
-     * @param {object} query user drill filter query
-     * @return {func} ajax func to request data and store in _data
-    */
-    countlySlippingPlugin.initialize = function(query) {
+/*global countlyCommon,jQuery,countlyVue*/
+(function(countlySlippingAwayUsers, $) {
 
-        //returning promise
-        return $.ajax({
-            type: "GET",
-            url: countlyCommon.API_URL + "/o/slipping",
-            data: {
-                app_id: countlyCommon.ACTIVE_APP_ID,
-                method: 'slipping',
-                query: JSON.stringify(query),
-            },
-            success: function(json) {
-                _data = json;
+    countlySlippingAwayUsers.helpers = {
+        buildFilters: function(filters) {
+            var newQuery = {};
+            if (filters.query) {
+                Object.keys(filters.query).forEach(function(queryItem) {
+                    var propertyValue = filters.query[queryItem];
+                    var propertyNameWithoutUp = queryItem.split('.')[1];
+                    newQuery[propertyNameWithoutUp] = propertyValue;
+                });
             }
-        });
+            return newQuery;
+        }
     };
 
-    countlySlippingPlugin.getData = function() {
-        return _data;
+    countlySlippingAwayUsers.service = {
+        fetchSlippingAwayUsers: function(filters, callback) {
+            $.ajax({
+                type: "GET",
+                url: countlyCommon.API_URL + "/o/slipping",
+                data: {
+                    app_id: countlyCommon.ACTIVE_APP_ID,
+                    method: 'slipping',
+                    query: JSON.stringify(countlySlippingAwayUsers.helpers.buildFilters(filters))
+                },
+                success: function(json) {
+                    callback(null, json);
+                },
+                error: function(error) {
+                    callback(error, null);
+                }
+            });
+        }
     };
 
-}(window.countlySlippingPlugin = window.countlySlippingPlugin || {}, jQuery));
\ No newline at end of file
+    countlySlippingAwayUsers.getVuexModule = function() {
+        var getInitialState = function() {
+            return {
+                slippingAwayUsers: [],
+                slippingAwayUsersFilters: { query: {}, byVal: []},
+                isLoading: false,
+                hasError: false,
+                error: null,
+            };
+        };
+
+        var slippingAwayUsersActions = {
+            fetchSlippingAwayUsers: function(context) {
+                context.dispatch('onFetchInit');
```

Comment:
Using "slippingAwayUsers" in action names (or mutations) feels a bit verbose.

Reply:
I, myself, have problems to succinctly use naming variables (count. working on it). Please let me know if you have any suggestions in this particular case. I must say, I am not fond of short, general variable names that lack context.

Reply:
Since countlySlippingAwayUsers already creates a context, everything defined under that is automatically namespaced. I also agree with you regarding using short and vague naming, but examples such as "countlySlippingAwayUsers/fetchSlippingAwayUsers" sounds/feels a bit verbose. Instead of that, for instance, we could use something like "countlySlippingAwayUsers/fetchAll".